clang/lib/Sema/SemaOverload.cpp

```diff
@@ -11757,6 +11791,42 @@ static void DiagnoseBadDeduction(Sema &S, NamedDecl *Found, Decl *Templated,
       return;
     }
 
+    // Errors in deduction guides from inherited constructors
+    // will manifest as substitution failures in the return type
+    // partial specialization, so we show a generic diagnostic
+    // in this case.
+    if (auto *DG = dyn_cast<CXXDeductionGuideDecl>(Templated);
```

Comment:
```suggestion
    if (const auto *DG = dyn_cast<CXXDeductionGuideDecl>(Templated);
```
and handle const correctness further down as well.

Reply:
Fixed